corehq/apps/export/models/deid_export.py

```diff
@@ -14,8 +15,7 @@ class DeIdHash(models.Model):
     def get_deid(cls, value, doc):
         domain = doc['domain'] if isinstance(doc, dict) else doc.domain
         salt = cls._get_salt(domain)
-        data = f'{value}{salt}'.encode('utf-8')
-        return md5(data).hexdigest()
+        return DeidGenerator(value, salt).random_hash()
```

Comment:
Got some feedback on the ticket that we should retain the original 10-character output format for these values. Just using the previous generator (with the addition of the new unique salt) felt like the nicest way to do that without reinventing the wheel.

Reply:
Would `md5(data).hexdigest()[:10]` also work? From my perspective it's far simpler.

Reply:
Not quite, since it's only hex (0-9, A-F) versus base 36 (0-9, A-Z). It is a subset of the expected format and would technically _work_ but I believe the possibility for hash collision would be much higher. I looked into approaches to get a hex or bytes hash into a base 36, and everything I found seemed like it would be essentially reinventing most of what `DeidGenerator` is already doing.

Reply:
Does a hash collision matter here? I think it may not.

For small numbers of hashes (n<1000), the probability is very low. For large numbers (approaching 1 million, for example), a collision does get significantly more likely (50%).

Reply:
I think occasional collisions probably do not matter in the case of DeID. Practically it means that two pieces of sensitive information that previously did not match could end up with the same value. But the object of DeID is to make sensitive information impossible to guess, and I think collisions actually help in that regard.

Another property of this DeID implementation is that common values (names like 'John' or an empty field) will always have the same hash for a given domain, and contextual information may make those values easier to guess. Having the possibility of collisions dilutes that ever so slightly.

Reply:
Couple of things here:
- got input from @kcowger that we do want to do our best to guarantee uniqueness of de-identified values even in very large exports.
- there is likely a bug with certain form exports in this code that I need to address, where either the `doc` passed in is missing or invalid for whatever reason and there's no `domain` to get from it. Based on current prioritization putting this on my back-burner, that'll likely get done mid-April.